telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
     failed_set[url] = output
     pickle.dump(failed_set, open('failed_urls', 'a'))
 
+def copy_wpr_to_benchmark():
+    """Moves wpr and .pc.wpr (all) files from local data/wpr_source to chromium
+
+    located: /src/tools/perf/page_sets/data/
+    """
+    local_path = os.path.join(os.getcwd(), 'data/wpr_source/*')
+    remote_data_path = os.path.join(CHROMIUM_SRC, 'tools/perf/page_sets/data/')
+    p = Popen('cp {0} {1}'.format(local_path, remote_data_path), shell=True)
+
+def prepare_benchmark(trial_number, num_urls):
+    """Simply checks to make sure the env is prepped before benchmarking
+
+    1. Ensures that ~/page_load_time/telemetry/temp/benchmark_results.db exists
+    2. Writes benchmarks to ~/src/tools/perf/benchmarks/page_cycler.py if they
+       do not already exist
+    NOTE: You may need to clear out the end of
+    ~/src/tools/perf/benchmarks/page_cycler.py for a clean start!
```

Comment:
Good idea - I ended up creating an entirely new benchmark file, telemetryBenchmarks, separate of page_cycler, but still using it's main class. That means I have to dynamically write out the class (shown below), but can simply re-write this file every time. This is a lot cleaner than checking for a string, and a lot simpler to debug.





```
from measurements import page_cycler                                               
import page_sets                                                                   
from telemetry import benchmark                                                    
                                                                                   
class _PageCycler(benchmark.Benchmark):                                            
    options = {'pageset_repeat': 6}                                                
    @classmethod                                                                   
    def AddBenchmarkCommandLineArgs(cls, parser):                                  
        parser.add_option('--v8-object-stats',                                     
            action='store_true',                                                   
            help='Enable detailed V8 object statistics.')                          
        parser.add_option('--report-speed-index',                                  
            action='store_true',                                                   
            help='Enable the speed index metric.')                                 
        parser.add_option('--cold-load-percent', type='int', default=50,           
            help='%d of page visits for which a cold load is forced')              
                                                                                   
    def CreatePageTest(self, options):                                             
        return page_cycler.PageCycler(                                             
            page_repeat = options.page_repeat,                                     
            pageset_repeat = options.pageset_repeat,                               
            cold_load_percent = options.cold_load_percent,                         
            record_v8_object_stats = options.v8_object_stats,                      
            report_speed_index = options.report_speed_index)  
```